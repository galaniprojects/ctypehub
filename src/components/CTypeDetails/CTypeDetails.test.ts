import { describe, expect, it } from '@jest/globals';

import { getSnapshotHtmlForPath } from '../../../testing/getSnapshotHtmlForPath';

describe('CTypeDetails', () => {
  it('should match snapshot', async () => {
    const html = await getSnapshotHtmlForPath('test/CTypeDetails-example');
    expect(html).toMatchSnapshot();
  });
  it('should handle nested CType property', async () => {
    const html = await getSnapshotHtmlForPath(
      'test/CTypeDetails-nestedProperty',
    );
    expect(html).toMatchSnapshot();
  });
  it('should handle nested CType', async () => {
    const html = await getSnapshotHtmlForPath('test/CTypeDetails-nestedCType');
    expect(html).toMatchSnapshot();
  });
});
