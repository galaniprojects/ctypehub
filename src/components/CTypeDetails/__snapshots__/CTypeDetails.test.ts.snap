// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CTypeDetails > should handle nested CType 1`] = `
<section class="_container">
  <h1 class="title">Example nested CType</h1>

  <dl>
    <dt>Creator:</dt>
    <dd>did:kilt:4pehddkhEanexVTTzWAtrrfo2R7xPnePpuiJLC7shQU894aY</dd>

    <dt>Description:</dt>
    <dd>This is an example of a CType with a nested CType</dd>

    <dt>Number of Attestations:</dt>
    <dd>333</dd>

    <dt id="/properties/example">Property 1:</dt>
    <dd><a href="/ctype/kilt:ctype:0x1">example</a></dd>

    <dt id="/properties/hasNestedCType">Property 2:</dt>
    <dd>hasNestedCType</dd>

    <div class="indent">
      <dt>Type:</dt>
      <dd>boolean</dd>
    </div>

    <dt>Creation date:</dt>
    <dd>5/1/2023, 12:00:00 PM</dd>

    <dt>Extrinsic:</dt>
    <dd>
      <a href="https://example.subscan.io/extrinsic/0xexamplehash">
        0xexamplehash
      </a>
    </dd>

    <dt>
      <a class="showDetails" href="#details">Technical Details:</a>
    </dt>

    <dd id="details">
      <pre>
{
  &quot;$id&quot;: &quot;kilt:ctype:0xea8670f43bb28356de7becdac900dd48f60a3d0d863c72b8fa11c841946529b6&quot;,
  &quot;$schema&quot;: &quot;ipfs://bafybeiah66wbkhqbqn7idkostj2iqyan2tstc4tpqt65udlhimd7hcxjyq/&quot;,
  &quot;additionalProperties&quot;: false,
  &quot;properties&quot;: {
    &quot;example&quot;: {
      &quot;$ref&quot;: &quot;kilt:ctype:0x1&quot;
    },
    &quot;hasNestedCType&quot;: {
      &quot;type&quot;: &quot;boolean&quot;
    }
  },
  &quot;title&quot;: &quot;Example nested CType&quot;,
  &quot;type&quot;: &quot;object&quot;
}</pre>
    </dd>
  </dl>
</section>
`;

exports[`CTypeDetails > should handle nested CType property 1`] = `
<section class="_container">
  <h1 class="title">Example nested CType property</h1>

  <dl>
    <dt>Creator:</dt>
    <dd>did:kilt:4pehddkhEanexVTTzWAtrrfo2R7xPnePpuiJLC7shQU894aY</dd>

    <dt>Description:</dt>
    <dd>This is an example of a CType with a nested property</dd>

    <dt>Number of Attestations:</dt>
    <dd>22</dd>

    <dt id="/properties/example">Property 1:</dt>
    <dd><a href="/ctype/kilt:ctype:0x1#/properties/example">example</a></dd>

    <dt id="/properties/hasNestedProperty">Property 2:</dt>
    <dd>hasNestedProperty</dd>

    <div class="indent">
      <dt>Type:</dt>
      <dd>boolean</dd>
    </div>

    <dt>Creation date:</dt>
    <dd>5/1/2023, 12:00:00 PM</dd>

    <dt>Extrinsic:</dt>
    <dd>
      <a href="https://example.subscan.io/extrinsic/0xexamplenestedhash">
        0xexamplenestedhash
      </a>
    </dd>

    <dt>
      <a class="showDetails" href="#details">Technical Details:</a>
    </dt>

    <dd id="details">
      <pre>
{
  &quot;$id&quot;: &quot;kilt:ctype:0x12501132c73da08f9aa80abcb2152f242de7b342c0e9a3e12fa8aacda35b89ea&quot;,
  &quot;$schema&quot;: &quot;ipfs://bafybeiah66wbkhqbqn7idkostj2iqyan2tstc4tpqt65udlhimd7hcxjyq/&quot;,
  &quot;additionalProperties&quot;: false,
  &quot;properties&quot;: {
    &quot;example&quot;: {
      &quot;$ref&quot;: &quot;kilt:ctype:0x1#/properties/example&quot;
    },
    &quot;hasNestedProperty&quot;: {
      &quot;type&quot;: &quot;boolean&quot;
    }
  },
  &quot;title&quot;: &quot;Example nested CType property&quot;,
  &quot;type&quot;: &quot;object&quot;
}</pre>
    </dd>
  </dl>
</section>
`;

exports[`CTypeDetails > should match snapshot 1`] = `
<section class="_container">
  <h1 class="title">Example CType</h1>

  <dl>
    <dt>Creator:</dt>
    <dd>did:kilt:4pehddkhEanexVTTzWAtrrfo2R7xPnePpuiJLC7shQU894aY</dd>

    <dt>Description:</dt>
    <dd>This is some example cType data</dd>

    <dt>Number of Attestations:</dt>
    <dd>1</dd>

    <dt id="/properties/example">Property 1:</dt>
    <dd>example</dd>

    <div class="indent">
      <dt>Type:</dt>
      <dd>string</dd>
    </div>
    <dt id="/properties/isExample">Property 2:</dt>
    <dd>isExample</dd>

    <div class="indent">
      <dt>Type:</dt>
      <dd>boolean</dd>
    </div>

    <dt>Creation date:</dt>
    <dd>5/1/2023, 12:00:00 PM</dd>

    <dt>Extrinsic:</dt>
    <dd>
      <a href="https://example.subscan.io/extrinsic/0xexamplehash">
        0xexamplehash
      </a>
    </dd>

    <dt>
      <a class="showDetails" href="#details">Technical Details:</a>
    </dt>

    <dd id="details">
      <pre>
{
  &quot;$id&quot;: &quot;kilt:ctype:0xdd49b562c6a808dce6ab1cac484405c1e02a917a178e923e4fcce3f9855834e4&quot;,
  &quot;$schema&quot;: &quot;ipfs://bafybeiah66wbkhqbqn7idkostj2iqyan2tstc4tpqt65udlhimd7hcxjyq/&quot;,
  &quot;additionalProperties&quot;: false,
  &quot;properties&quot;: {
    &quot;example&quot;: {
      &quot;type&quot;: &quot;string&quot;
    },
    &quot;isExample&quot;: {
      &quot;type&quot;: &quot;boolean&quot;
    }
  },
  &quot;title&quot;: &quot;Example CType&quot;,
  &quot;type&quot;: &quot;object&quot;
}</pre>
    </dd>
  </dl>
</section>
`;
